cmd/containerd-shim-runhcs-v1/task_jobobject_update.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/Microsoft/hcsshim/internal/jobobject"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/processorinfo"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	jobObjUtilExeName = "jobobject-util.exe"
+
+	cpuLimitFlag     = "cpu-limit"
+	cpuWeightFlag    = "cpu-weight"
+	useNTVariantFlag = "use-nt"
+)
+
+func createJobObjectsUtilArgs(ctx context.Context, toolPath string, procResources *specs.WindowsCPUResources) []string {
+	args := []string{"cmd", "/c", toolPath, "set", useNTVariantFlag}
+	if procResources.Count != nil {
+		procCount := uint32(*procResources.Count)
+		hostProcs := uint32(processorinfo.ProcessorCount())
```

Comment:
This is going to give us the processor count of the host. But the number of processor cores available to the UVM could be different right? In that case do we want to check against the cores available to the UVM?

Reply:
ah good catch. I'll fix this.

Reply:
fixed.